### PR TITLE
Implement suggest command

### LINKED ISSUE #48: Implement suggest command

### PR DESCRIPTION
**Pull request by Hilda-Ang**
Resolves #48

### CODE REVIEW

**Comment by Hilda-Ang** [comment]
@rongjiecomputer I add back the constructor `ModuleInfo(String code)` because many tests will break otherwise

**Comment by rongjiecomputer** [comment]
> @rongjiecomputer I add back the constructor `ModuleInfo(String code)` because many tests will break otherwise

Ok, I will check the tests that use this constructor and see if I can make them don't use it.

File: src/main/java/seedu/planner/model/ModulePlanner.java [hunk 1]
```diff
@@ -107,25 +109,29 @@ public boolean hasModule(Module module) {
...
-     *                      are stored at
      * @return An {@code ObservableList} containing all the {@code Module}s
      */
     @Override
-    public ObservableList<Module> listModulesAvailable(int index) {
-        return FXCollections.unmodifiableObservableList(
-                semesters.get(index).getModulesAvailable());
+    public ObservableList<Module> getModulesAvailable() {
+        List<Module> modulesAvailable = new ArrayList<>();
+        List<Module> modulesTaken = this.getAllModulesTaken();
```
Comment (GabrielYik) [inline src/main/java/seedu/planner/model/ModulePlanner.java]: Can remove the `this`s to follow coding standard

File: src/main/java/seedu/planner/model/ReadOnlyModulePlanner.java [hunk 2]
```diff
@@ -24,13 +24,12 @@
      * @param index The nominal semester index the {@code Module}s are stored at
```
Comment (GabrielYik) [inline src/main/java/seedu/planner/model/ReadOnlyModulePlanner.java]: Can remove mention of `nominal`

**Pull request merged**